src/Libraries/RevitNodesUI/Selection.cs

```diff
@@ -155,8 +175,11 @@ public override void Dispose()
 
             if (revitDynamoModel != null)
             {
-                var hwm = RevitDynamoModel.Workspaces.OfType<HomeWorkspaceModel>().ElementAt(0);
-                hwm.RunSettings.PropertyChanged -= revMod_PropertyChanged;
+                var hwm = revitDynamoModel.CurrentWorkspace as HomeWorkspaceModel;
+                if (hwm != null)
+                {
+                    hwm.RunSettings.PropertyChanged -= revMod_PropertyChanged;
```

Comment:
I see that the de-registration is done here. Then why is it needed again in `RevitDynamoModel` setter?